tests/test__collection_api.py

```diff
@@ -1806,3 +1806,37 @@ def test_aggregate_unwind_push_first(self):
                 '_id': 'ooo'
             }]
         self.assertEqual(expect, list(actual))
+
+    def test__aggregate_group_scalar_key(self):
+        collection = self.db.collection
+        collection.insert_many(
+            [
+                {'a': 2, 'b': 3, 'c': 4},
+                {'a': 2, 'b': 3, 'c': 5},
+                {'a': 1, 'b': 1, 'c': 1},
+            ]
+        )
+        actual = collection.aggregate([
+            {'$group': {'_id': '$a'}},
+        ])
+        self.assertEqual(
+            [{'_id': 1}, {'_id': 2}],
+            list(actual)
+        )
+
+    def test__aggregate_group_dict_key(self):
+        collection = self.db.collection
+        collection.insert_many(
+            [
+                {'a': 2, 'b': 3, 'c': 4},
+                {'a': 2, 'b': 3, 'c': 5},
+                {'a': 1, 'b': 1, 'c': 1},
+            ]
+        )
+        actual = collection.aggregate([
+            {'$group': {'_id': {'a': '$a', 'b': '$b'}}},
+        ])
+        self.assertEqual(
+            [{'_id': {'a': 1, 'b': 1}}, {'_id': {'a': 2, 'b': 3}}],
```

Comment:
I agree your code is doing exactly that, however it's not clear that this is how MongoDB would sort it. Especially if one item of the collection is missing a b field. When I test on a real MongoDB, `{'a': 2}` gets sorted before `{'a': 1, 'b': 1}` and that wouldn't be the case with your code. Can you please add a test in `test__mongomock.py` to compare the mongomock behaviour with the one from pymongo + MongoDB please?

Reply:
Indeed, you're right! So the sorting is only useful to use itertools.groupby, it makes total sense.

Can you please add a python comment just above the `out_collection = sorted(out_collection, key=sort_key_getter)` line explaining that (so that the next dev that needs to fix it don't get it wong):

```py
# Sort the collection only for the itertools.groupby. $group does not order its output document.
```

Also maybe add a python comment in the tests to make sure that if someone changes the order and the tests break, they don't panick and just update the test. Bonus points if you can make the tests not enforce the current order (and then drop the comment in the test).

Reply:
Good idea. thx. Updated now.